mountpoint-s3/src/upload.rs

```diff
@@ -88,6 +94,7 @@ impl<Client: ObjectClient> UploadRequest<Client> {
             key: key.to_owned(),
             next_request_offset: 0,
             hasher: Hasher::new(),
```

Comment:
The `trailing_checksums` flag should also affect hashing: if disabled, we don't need the `Hasher`, or at least we don't need to calculate the checksum in `write`.